### PR TITLE
Git plugin conflict with gdc (GNU D Compiler)

### PR DESCRIPTION
**Pull request by CorvinM**
Added check for gdc compiler before adding the conflicting gdc alias in the git plugin

### CODE REVIEW

File: plugins/git/git.plugin.zsh [hunk 1]
```diff
@@ -80,7 +80,12 @@ alias gcp='git cherry-pick'
 alias gcs='git commit -S'
 
 alias gd='git diff'
+
+#Dont add alias if gdc (GCC D Compiler) is present
+command -v gdc >/dev/null 2>&1 || {
```
Comment (blueyed) [inline plugins/git/git.plugin.zsh]: You can use zsh's `$commands` array, e.g. `if (( $+commands[gdc] )); then …`, or `if [[ -n "${commands[gdc]}" ]]`.

That should be provided through a `safe_alias` function though (see https://github.com/robbyrussell/oh-my-zsh/issues/3682#issuecomment-79344414).

**Comment by blueyed** [comment]
I think this PR can be closed in favour of #3990.

**Comment by ncanceill** [comment]
Wait, we yet have to agree on a replacement alias and include it in #3990.

**Comment by ncanceill** [comment]
But yes, this will be closed by #3990 anyway.

**Comment by juanpastas** [comment]
I don't understand, this means that if I have if gdc is not showing git diff it should show some thing about gcc d compiler?

I don't have that compiler, but still I need to use gdca, is that correct?

**Comment by blueyed** [comment]
@juanpastas 
The idea is to not setup a `gdc` alias if there is a `gdc` program.

(your question is not really clear to me however)

**Comment by juanpastas** [comment]
What I meant is that gdc says command not found, while instead of that it
should output cached diff.

But this does not happen because it was just changed to gdca, but not
actually detecting if gdc overlaps other command.

**Comment by blueyed** [comment]
Yes, it was changed for now, but might come back.
See https://github.com/robbyrussell/oh-my-zsh/pull/3990#issuecomment-111175759 and following comments.

**Comment by juanpastas** [comment]
Thanks.

**Pull request closed without merging**